Apromore-Core-Components/Apromore-Manager/src/main/java/org/apromore/service/impl/WorkspaceServiceImpl.java

```diff
@@ -695,26 +697,125 @@ private void removeGroupLog(Group group, Log log) {
         }
     }
 
-    // TODO: whether should I put this logic here or in a separate service?
-//    private void createGroupUsermetadata(Group group, UserMetadata userMetadata, boolean hasRead, boolean hasWrite,
-//                                         boolean hasOwnership) {
-//        GroupUserMetadata groupUserMetadata = groupUserMetadataRepo.findByGroupAndUserMetadata(group, userMetadata);
-//        if (groupUserMetadata == null) {
-//            groupLog= new GroupLog(group, log, hasRead, hasWrite, hasOwnership);
-//            log.getGroupLogs().add(groupLog);
-//            //group.getGroupLogs().add(groupLog);
-//        } else {
-//            groupLog.setHasRead(hasRead);
-//            groupLog.setHasWrite(hasWrite);
-//            groupLog.setHasOwnership(hasOwnership);
-//        }
-//        groupLogRepo.save(groupLog);
-//    }
-//
-//    private void removeGroupUsermetadata(Group group, UserMetadata usermetadata) {
-//        GroupLog groupLog = groupLogRepo.findByGroupAndLog(group, log);
-//        if (groupLog != null) {
-//            groupLogRepo.delete(groupLog);
-//        }
-//    }
+    public Map<Group, AccessType> getLogACL(Integer logId) {
+
+        Map<Group, AccessType> groupAccessTypeMap = new HashMap<>();
+
+        for (GroupLog g : getGroupLogs(logId)) {
+            if (g.getHasRead() && g.getHasWrite() && g.getHasOwnership()) {
```

Comment:
This method should be renamed as: getLogAccessType() as shown from its implementation, the return is AccessType. ACL is Access Control List which is not at this level.